### PR TITLE
feat(hooks): add mcp_namespace_hint hook for @server_name references

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- Adds a new `mcp_namespace_hint` hook that fires on `GENERATION_PRE` when the last user message contains `@server_name` references matching loaded MCP servers
- Injects a compact system message listing available tools with their full `mcp__server__tool` names, so the model can pick the right tool without memorizing the flat namespace
- Mirrors Gemini CLI's `@`-mention UX for MCP server addressing

## Behaviour

```
User: @github list my open PRs

→ System: MCP tool hint for @github: 5 available tool(s). Use the full tool name (e.g. ``mcp__github__tool_name``).

- `mcp__github__list_issues`: List GitHub issues
- `mcp__github__list_pulls`: List pull requests
...
```

Hook only fires when:
- The last user message has `@name` references
- At least one reference matches a loaded MCP server
- That server has tools registered

## Test plan

- [x] 13 unit tests: regex, edge cases (no match, empty messages, no servers loaded), multi-server, tool-without-description, hint format
- [x] `make typecheck` clean
- [x] `uv run pytest tests/test_hooks_mcp_namespace_hint.py -q` — 13 passed

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
Addressed both Greptile review points:

- **Regex false-positive on emails**: Added `(?<!\w)` negative lookbehind so `user@github.com` no longer matches `github` when a GitHub MCP server is loaded. Added `test_no_false_positive_on_email_address` to lock this in.
- **Private import of `_mcp_clients`/`_dynamic_servers`**: Added `get_mcp_clients()` as a stable public accessor to `mcp_adapter`; the hook now imports that instead of the private dicts.

**Comment by TimeToBuildBob** [comment]
Addressed both Greptile review items:

1. **Regex false-positive on email addresses** - added negative lookbehind to `_MCP_HINT_RE`, with a regression test
2. **Private import of `_mcp_clients`/`_dynamic_servers`** - added `get_mcp_clients()` public accessor to mcp_adapter.py, hook now imports that instead

Committed as 5ccbebada + 2f36c42b4

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/hooks/mcp_namespace_hint.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        lines.append(
+            f"\nMCP tool hint for @{server_name}: {total} available tool(s). "
+            f"Use the full tool name (e.g. ``mcp__{server_name}__tool_name``).\n"
+        )
+        for tool in tools:
+            desc = getattr(tool, "description", "") or ""
+            name = getattr(tool, "name", "?")
+            desc_line = f"- `mcp__{server_name}__{name}`" + (
+                f": {desc}" if desc else ""
+            )
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/hooks/mcp_namespace_hint.py]: Fixed in 2f1c52b1c — updated both the hook and all test assertions to use `server.tool_name` dot-notation, matching how `mcp_adapter.py` registers tools at line 208.

**Comment by TimeToBuildBob** [comment]
@greptileai review
